app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb

```diff
@@ -1,6 +1,11 @@
 class ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Runner <
   ManageIQ::Providers::BaseManager::EventCatcher::Runner
-  INCLUDE_EVENTS = {'Hawkular Deployment' => '.*'}.freeze
+
+  # INCLUDE only supported events. Filter out any others.
+  # - event category must eq a hash key
+  # - event message must match hash value :pattern, if supplied
+  # - hash value :type specifies the entity type for which the event's canonical path must have an entry
+  INCLUDE_EVENTS = {'Hawkular Deployment' => {:pattern => '.*', :type => 'MiddlewareServer'}}.freeze
```

Comment:
there is a lot of knowledge in this hash.
and it actually has 2 responsibilties. One is whitelisting and one is mapping the type.
how about making it private instance variables which get setup in intialize?
```ruby
class 
def initialize
  super
  @whitelist_events = {
     'Hawkulare Deployment' => /.*/
  }
end

def valid?(event)
  return false unless @whitelist_events[event.category]
  @whitelist_events[event.category].match event.context['Message']
end
```

The type mapping should move to the `EventParser`